### ISSUE TITLE
azure_rm inventory plugin does not work with collection azure.azcollection 2.0.0 or 2.1.1

### ISSUE THREAD

**Issue by emilbra** [ansible/ansible_collections_azure#1384]
##### SUMMARY

While developing a new ansible project, we are attempting to use this very simple inventory:
```
plugin: azure_rm
include_vm_resource_groups:
  - my_rg
auth_source: auto
```

Using azure.azcollection versions 2.0.0 or 2.1.1 produces an error, see below
##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

azure_rm inventory plugin

##### ANSIBLE VERSION

```paste below
    2.15.4
```

##### COLLECTION VERSION

```paste below
# /Users/ebraso/.ansible/collections/ansible_collections
Collection         Version
------------------ -------
azure.azcollection 2.1.0  
```

##### CONFIGURATION

```paste below
CONFIG_FILE() = None
PAGER(env: PAGER) = less
```

##### OS / ENVIRONMENT

MacOS 13.6 (22G120)

##### STEPS TO REPRODUCE

Install azure.azcollection from ansible galaxy using `ansible-galaxy collection install azure.azcollection`

Run an ansible playbook against a host, such as below

```yaml
---

- name: print statement
  hosts: devel-mc-server_08a3
  gather_facts: false

  tasks:

    - name: Print
      ansible.builtin.debug:
        msg: "test"

```

##### EXPECTED RESULTS

Expected debug module to work without issue

##### ACTUAL RESULTS

while not using specifically version 2.1.0, any playbook fails with this error:
```
PLAY [print statement] ****************************************************************************************

TASK [Print] **************************************************************************************************
fatal: [devel-mc-server_08a3]: FAILED! => {"msg": "Invalid type for configuration option plugin_type: connection plugin: ssh setting: host (from var: ansible_host): Invalid type provided for \"string\": ['THEIP OF MY SERVER']"}

PLAY RECAP ****************************************************************************************************
devel-mc-server_08a3       : ok=0    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   
```

My solution was to specifically install version 2.1.0 of the azure.azcollection collection from ansible galaxy.
I have tested 2.0.0, which did not work, but have not tested even earlier versions.
I have not been able to find duplicate issues to this

**Comment by alessandrovalentiniwp** [comment]
Hi, it seems like we have the same issue after the upgrade of `azcollection` from version 2.1.0 to 2.1.1

In particular we are using the following a dynamic inventory:
```
plugin: azure.azcollection.azure_rm
use_extra_vars: true
auth_source: auto

include_vm_resource_groups:
  - {{ azure_resource_group }}

include_vmss_resource_groups:
  - '*'

plain_host_names: true

exclude_host_filters:
  - tags['Environment'] is not defined
  - tags['Environment'] != '{{ azure_tag_value }}'

conditional_groups:
  azure: true
```

but we get the following error:
```
myazurevm | UNREACHABLE! => {
    "changed": false,
    "msg": "Failed to connect to the host via ssh: ssh: Could not resolve hostname ['11.22.33.44']: Name or service not known",
    "unreachable": true
}
```

It seems that now `ansible_host` is now set to an IP array instead of a single IP like it was until 2.1.0 version. 
Downgrading `azcollection` from 2.1.1 to 2.1.0 solved the problem. 
In the case this is the intended behavior how can we pick the first IP of the list as `ansible_host` since our machine have only a single IP?

Is it possible that the bug has been introduced with the bugfix related to this issue? https://github.com/ansible-collections/azure/issues/1378

**Comment by Fred-sun** [comment]
> Hi, it seems like we have the same issue after the upgrade of `azcollection` from version 2.1.0 to 2.1.1
> 
> In particular we are using the following a dynamic inventory:
> 
> ```
> plugin: azure.azcollection.azure_rm
> use_extra_vars: true
> auth_source: auto
> 
> include_vm_resource_groups:
>   - {{ azure_resource_group }}
> 
> include_vmss_resource_groups:
>   - '*'
> 
> plain_host_names: true
> 
> exclude_host_filters:
>   - tags['Environment'] is not defined
>   - tags['Environment'] != '{{ azure_tag_value }}'
> 
> conditional_groups:
>   azure: true
> ```
> 
> but we get the following error:
> 
> ```
> myazurevm | UNREACHABLE! => {
>     "changed": false,
>     "msg": "Failed to connect to the host via ssh: ssh: Could not resolve hostname ['11.22.33.44']: Name or service not known",
>     "unreachable": true
> }
> ```
> 
> It seems that now `ansible_host` is now set to an IP array instead of a single IP like it was until 2.1.0 version. Downgrading `azcollection` from 2.1.1 to 2.1.0 solved the problem. In the case this is the intended behavior how can we pick the first IP of the list as `ansible_host` since our machine have only a single IP?
> 
> Is it possible that the bug has been introduced with the bugfix related to this issue? #1378

Fixed by #1379

**Comment by Fred-sun** [comment]
@emilbra It maybe fixed by #1379, Please try! Thank you very much!

**Comment by camillemahaut** [comment]
Hi have the issue like emilbra.
we moved from 1.19 to 2.1.1 of azcollection.
error:
"msg": "Invalid type for configuration option plugin_type: connection plugin: ssh setting: host (from var: ansible_host): Invalid type provided for \"string\": []"

Back to previous version 2.0.0 or 1.19, but I cannot use a list of resource group on this one, I only can use one resource group.

**Comment by camillemahaut** [comment]
ansible [core 2.16.2]
azure.azcollection 2.1.1

If i run : ` ansible-inventory -i /etc/ansible/azure_rm.yml --list`

this is the result 
```
"hostvars": {
"AAAAAAAAAA": {
"ansible_host": [],
"availability_zone": null,
"computer_name": "AAAAAAAAAAA",....
```
So ansible_host is empty, I have no public ip on my hosts.

**Comment by Fred-sun** [comment]
> ansible [core 2.16.2] azure.azcollection 2.1.1
> 
> If i run : ` ansible-inventory -i /etc/ansible/azure_rm.yml --list`
> 
> this is the result
> 
> ```
> "hostvars": {
> "AAAAAAAAAA": {
> "ansible_host": [],
> "availability_zone": null,
> "computer_name": "AAAAAAAAAAA",....
> ```
> 
> So ansible_host is empty, I have no public ip on my hosts.

@camillemahaut Thanks for you feedback! As described in the plugin/inventory/azure_rm.py document, The ansibe_host will be set to public_ip or private_ip if public_ip does not exist. This is correct. But now the return is empty, it is indeed a bug, has been fixed in #1406. Thank you!
